boa_engine/src/vm/mod.rs

```diff
@@ -180,8 +180,9 @@ impl Context<'_> {
 
         let completion = loop {
             let completion = self.run_next_instruction(|opcode, context| {
-                total_cost = total_cost.checked_add(opcode.cost()).unwrap_or(usize::MAX);
-                total_cost += opcode.cost();
+                total_cost = total_cost
+                    .checked_add(usize::from(opcode.cost()))
+                    .unwrap_or(usize::MAX);
```

Comment:
[`.saturating_add()`](https://doc.rust-lang.org/std/primitive.u32.html#method.saturating_add) would achieve the same thing without the `unwrap_or`